pkg/prowgen/prowgen.go

```diff
@@ -218,6 +217,41 @@ func generatePostsubmitForTest(jobBaseBuilder *prowJobBaseBuilder, info *Prowgen
 	}
 }
 
+func generatePostsubmitsForPromotion(jobBaseBuilderFactory func() *prowJobBaseBuilder, info *ProwgenInfo, options ...generatePostsubmitOption) ([]prowconfig.Postsubmit, error) {
+	opts := &generatePostsubmitOptions{}
+	for _, opt := range options {
+		opt(opts)
+	}
+	architectures := append([]api.Architecture{api.AMD64Arch}, info.Config.AdditionalArchitectures...)
+	postsubmits := make([]prowconfig.Postsubmit, 0, len(architectures))
+	for _, arch := range architectures {
+		jobBaseBuilder := jobBaseBuilderFactory()
+		var jobBaseGen *prowJobBaseBuilder
+		if arch != api.AMD64Arch {
+			testName := fmt.Sprintf("images-%s", string(arch))
+			cluster, found := arch.GetMappedCluster()
+			if !found {
```

Comment:
You will never hit this since you are validating the config.

Reply:
Yes, but this is just an execution path, we should not ignore the error because we know we have validated the config beforehand.